src/main/java/seedu/address/model/person/FamilySize.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.model.person;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's family size (inclusive of himself) in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidFamilySize(int)}
+ */
+public class FamilySize {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Family size should be a non-negative integer";
+
+    private final int value;
+
+    /**
+     * Constructs a {@code FamilySize}.
+     *
+     * @param size A valid size integer
+     */
+    public FamilySize(int size) {
+        checkArgument(isValidFamilySize(size), MESSAGE_CONSTRAINTS);
```

Comment:
Is it necessary to check the isValidFamilySize condition here again since it is already being checked by the same method in ParserUtil during parsing before being allowed through to the FamilySize constructor.

Reply:
This argument check is necessary here as this constructor might be called by other classes. This check enforces the validity of the argument.